### PR TITLE
#130 Modify load to become more generic work and make it abstract

### LINKED ISSUE #130: Modify "Load" to become "Work" and make it abstract
This is so that the concept of "Load" might be made to integrate non-loaded based metrics, in particular using communication.

### PR DESCRIPTION
**Pull request by ppebay**
Fixes #130

### CODE REVIEW

File: src/Applications/LBAF.py [hunk 1]
```diff
@@ -149,7 +152,11 @@ def get_conf_file(self, conf_file=os.path.join(project_path, "src", "Application
         """
 
         if os.path.splitext(conf_file)[-1] in [".yml", ".yaml"] and os.path.isfile(conf_file):
-            print(f"{bcolors.OKMSG}Config file {conf_file} FOUND!{bcolors.END}")
+            # Try to open configuration file
+            print(bcolors.HEADER
```
Comment (marcinwrobel1986) [inline src/Applications/LBAF.py:156]: Maybe `fstring` here?
```Python
print(f"{bcolors.HEADER}[LBAF]{bcolors.END} Found configuration file {conf_file}")
```

Comment (ppebay) [reply src/Applications/LBAF.py:156]: Yeah why not. But I don't like the `fstring` syntax. I find it much less clear.

Comment (ppebay) [reply src/Applications/LBAF.py]: Done

File: src/Execution/lbsRuntime.py [hunk 2]
```diff
@@ -76,13 +78,23 @@ def __init__(self, p, c: dict, order_strategy: str, v=False):
...
             self.phase = p
 
-        # Verbosity of runtime
-        self.verbose = v
+        # Instantiate work model
+        self.work_model = WorkModelBase.factory(
+            w.get("name"),
+            w.get("parameters", {}))
+        if not self.work_model:
+            print(bcolors.ERR
```
Comment (marcinwrobel1986) [inline src/Execution/lbsRuntime.py:86]: So, maybe we switch to fstring with every new commit :)?

File: src/Execution/lbsRuntime.py [hunk 3]
```diff
@@ -93,21 +105,11 @@ def __init__(self, p, c: dict, order_strategy: str, v=False):
         _, l_min, self.average_load, l_max, l_var, _, _, l_imb = compute_function_statistics(
             self.phase.ranks,
             lambda x: x.get_load())
-        n_w, _, w_ave, w_max, w_var, _, _, w_imb = compute_function_statistics(
+        n_w, _, w_ave, w_max, _, _, _, _ = compute_function_statistics(
```
Comment (marcinwrobel1986) [inline src/Execution/lbsRuntime.py:108]: Just 3 out of 8 function outputs in use, maybe it's time to modify the output of `compute_function_statistics`

Comment (ppebay) [reply src/Execution/lbsRuntime.py:108]: Yes. We should return a structure and create a class for that (`lbsStatisticsContainer` or something like that). Another day. Please create another issue for that. Thanks.

File: src/Execution/lbsRuntime.py (hunk 2 above)
Comment (ppebay) [reply src/Execution/lbsRuntime.py:86]: Better yet we should have a helper function called `print_error()`, another one called `print_warning()` and yet another one called `print_message()` that would generically deal with the properly-colored headings. In addition I would make the warnings have an orange-colored heading, the regular messages have a green heading (and keep the red for errors).

File: src/IO/lbsLoadReaderVT.py [hunk 4]
```diff
@@ -193,7 +193,10 @@ def json_reader(self, returned_dict: dict, file_name: str, phase_ids, node_id: i
 
         # validate schema
         if SchemaValidator().is_valid(schema_to_validate=decompressed_dict):
-            print(f"{bcolors.OK}[LoadReaderVT] Valid JSON schema in {file_name}{bcolors.END}")
+            print(bcolors.HEADER
```
Comment (marcinwrobel1986) [inline src/IO/lbsLoadReaderVT.py:196]:
```Python
print(f"{bcolors.HEADER}[LoadReaderVT]{bcolors.END} Valid JSON schema in {file_name}")
```

Comment (ppebay) [reply src/IO/lbsLoadReaderVT.py:196]: Last one I am changing.
After that please check out #132

**Review by marcinwrobel1986 — APPROVED** [review]

**Pull request merged**